src/Data/Codec/JSON.purs

```diff
@@ -215,25 +219,55 @@ type PropCodec a =
 -- | See also `Data.Codec.JSON.Record.object` for a more commonly useful
 -- | version of this function.
 object ∷ ∀ a. PropCodec a → Codec a
-object codec =
-  Codec.codec'
-    (\j → Codec.decode codec =<< Codec.decode jobject j)
-    (\a → encode jobject (JO.fromFoldable (encode codec a)))
+object = object' { ignoreUnknownProps: true }
+
+-- | A version of the `object` function that allows to specify whether unknown
+-- | properties should be ignored or cause an error.
+-- |
+-- | See also `Data.Codec.JSON.Record.object'` for a more commonly useful
+-- | version of this function.
+object' ∷ ∀ a. { ignoreUnknownProps :: Boolean } -> PropCodec a -> Codec a
+object' { ignoreUnknownProps } codec = Codec.codec' dec enc
```

Comment:
Agreed with Gary that the implementation for this should be separate from the lax case, at which point we can avoid the additional parameter (since it would always be `false`) and name it in a way that more clearly expresses the intent, e.g. `objectStrict`